tests/includes/class-vip-enterprise-search-unit-test-case.php

```diff
@@ -6,53 +6,71 @@
  * @subpackage Tests
  */
 
+use \ElasticPress\Elasticsearch;
+use \ElasticPress\Indexables;
+
 /**
  * VIP_Enterprise_Search_Adapter_UnitTestCase class.
  */
 class VIP_Enterprise_Search_Adapter_UnitTestCase extends Adapter_UnitTestCase {
 
-	public static function set_up_before_class() {
-		parent::set_up_before_class();
+	public static function setUpBeforeClass(): void {
+		parent::setUpBeforeClass();
 		static::flush();
+
+		// Generate the mapping for the post index.
+		$indexable = Indexables::factory()->get( 'post' );
+		$index_name = $indexable->get_index_name();
+		$mapping = $indexable->generate_mapping();
+		Elasticsearch::factory()->put_mapping( $index_name, $mapping );
+
+		// Create and index posts.
+		$posts = self::create_sample_content();
+		self::index_content( $posts );
 	}
 
-	public static function tear_down_after_class() {
-		static::flush();
-		parent::tear_down_after_class();
+	protected function tearDown(): void {
+		// Reset to default state. Includes features and config as well.
+		self::flush();
+		parent::tearDown();
 	}
 
 	/**
 	 * Flush the index.
-	 *
-	 * TODO implement.
+	 * Deletes the index created for the tests.
 	 *
 	 * @see See Adapter_UnitTestCase::flush()
 	 */
 	protected static function flush(): void {
+		$ep = new Elasticsearch();
+		$ep->delete_index( Indexables::factory()->get( 'post' )->get_index_name() );
 	}
 
 	/**
 	 * Force Elasticsearch to refresh its index to make content changes
 	 * available to search.
 	 *
-	 * TODO implement
-	 *
 	 * @see See Adapter_UnitTestCase::refresh_index()
 	 */
 	protected static function refresh_index(): void {
+		$ep = new Elasticsearch();
+		$ep->refresh_indices();
 	}
 
 	/**
 	 * Index one or more posts in Elasticsearch.
 	 *
-	 * TODO implement
-	 *
 	 * @see See Adapter_UnitTestCase::index()
 	 * @see See Adapter_UnitTestCase::index_content()
 	 *
 	 * @param mixed $posts Can be a post ID, WP_Post object, or
-	 *                     an array of any of the above.
+	 *                     an array of either of the above.
 	 */
 	protected static function index_content( $posts ): void {
+		Indexables::factory()->get( 'post' )->bulk_index( $posts );
+
+		// This isn't optimal, but the indexing requires time to complete, without it,
+		// the tests run before the index operation is complete. Time depends on number of posts being indexed.
+		sleep( 2 );
```

Comment:
@kevinfodness / @mboynes This is another piece that leaves a bad taste in my mouth, but I can't think of a better approach that wouldn't be over-engineered. The comment above says it all. Do y'all have any ideas about how to approach this in another way?

Reply:
You could write a loop with a set number of retries that queries for the posts in the index, and re-runs the loop up to N number of times until the item appears in the index. We [do something similar in SearchPress](https://github.com/alleyinteractive/searchpress/blob/main/tests/bootstrap.php#L29) which balances the need to wait for something to happen with the fact that it may not happen in the same amount of time all of the time and needing to put some kind of a sensible cap on it.

Reply:
I created https://github.com/alleyinteractive/elasticsearch-extensions/issues/76 from this comment